src/components/CouponDetails/index.jsx

```diff
@@ -281,23 +313,83 @@ class CouponDetails extends React.Component {
     });
   }
 
+  updateSelectAllCheckBox() {
+    const { selectedCodes, tableColumns } = this.state;
+    const { couponDetailsTable: { data: tableData } } = this.props;
+
+    const allCodesForPageSelected = tableData && selectedCodes.length === tableData.results.length;
+    const hasPartialSelection = selectedCodes.length > 0 && !allCodesForPageSelected;
+
+    const selectColumn = tableColumns.shift();
+
+    selectColumn.label = React.cloneElement(selectColumn.label, {
+      checked: allCodesForPageSelected,
+      className: hasPartialSelection ? ['mixed'] : [],
+    });
+
+    // The Paragon `CheckBox` component does not currently support the mixed state. To
+    // get around this, we get the DOM node of the checkbox and replace the `aria-checked`
+    // attribute appropriately.
+    //
+    // TODO: We may want to update Paragon `CheckBox` component to handle mixed state.
+    const selectAllCheckBoxRef = selectColumn.label.ref && selectColumn.label.ref.current;
+    const selectAllCheckBoxDOM = (
+      selectAllCheckBoxRef && document.getElementById(selectAllCheckBoxRef.state.id)
+    );
+
+    if (selectAllCheckBoxDOM && hasPartialSelection) {
+      selectAllCheckBoxDOM.setAttribute('aria-checked', 'mixed');
+    } else if (selectAllCheckBoxDOM && !hasPartialSelection) {
+      selectAllCheckBoxDOM.setAttribute('aria-checked', allCodesForPageSelected);
+    }
+
+    this.setState({
+      tableColumns: [selectColumn, ...tableColumns],
+    });
+  }
+
   handleCodeSelection({ checked, code }) {
-    const { selectedCodes } = this.state;
+    let { selectedCodes, hasAllCodesSelected } = this.state;
 
     if (checked) {
       // Add code to selected codes array
-      this.setState({
-        selectedCodes: [...selectedCodes, code.code],
-      });
+      selectedCodes = [...selectedCodes, code];
     } else {
       // Remove code from selected codes array
-      this.setState({
-        selectedCodes: selectedCodes.filter(codeTitle => codeTitle !== code.code),
-      });
+      selectedCodes = selectedCodes.filter(selectedCode => selectedCode.code !== code.code);
```

Comment:
the `code` versus `code.code` is pretty confusing in this method - especially since selectedCodes stores the enveloped `code`. Maybe just a comment to explain whats in selectedCodes array versus the actual code in `code.code`

Reply:
@georgebabey This could be changed to:
```js
selectedCodes = selectedCodes.filter(selectedCode => selectedCode !== code)
```
Would that be clearer?

Reply:
Yes, I think that is easier to read 👍